examples/govpp_call/main.go

```diff
@@ -48,103 +51,96 @@ import (
 // HTTP and Log), GOVPP, resync plugin and example plugin which demonstrates GOVPP call functionality.
 func main() {
 	// Init close channel to stop the example
-	closeChannel := make(chan struct{}, 1)
+	exampleFinished := make(chan struct{}, 1)
+
+	// Start Agent with ExampleFlavor (combinatioplugin.GoVppmux, n of ExamplePlugin & reused cn-infra plugins)
+	flavor := ExampleFlavor{closeChan: &exampleFinished}
+	agent := core.NewAgent(log.DefaultLogger(), 15*time.Second, append(flavor.Plugins())...)
+	core.EventLoopWithInterrupt(agent, exampleFinished)
+}
 
-	f := vpp.Flavor{}
+/**********
+ * Flavor *
+ **********/
 
-	// Example plugin (GOVPP call)
-	examplePlugin := ExamplePlugin{}
-	examplePlugin.GoVppmux = &f.GoVPP
+// ETCD flag to load config
+func init() {
+	flag.String("etcdv3-config", "etcd.conf",
+		"Location of the Etcd configuration file")
+}
 
-	namedExamplePlugin := &core.NamedPlugin{PluginName: PluginID, Plugin: &examplePlugin}
+// ExampleFlavor is a set of plugins required for the datasync example.
+type ExampleFlavor struct {
+	// Local flavor to access to Infra (logger, service label, status check)
+	*vpp.Flavor
+	// Example plugin
+	GovppExample ExamplePlugin
+	// For example purposes, use channel when the example is finished
+	closeChan *chan struct{}
+}
 
-	// Create new agent
-	agent := core.NewAgent(log.DefaultLogger(), 15*time.Second, append(f.Plugins(), namedExamplePlugin)...)
+// Inject sets object references
+func (ef *ExampleFlavor) Inject() (allReadyInjected bool) {
+	// Init local flavor
```

Comment:
please put to every Inject() method:

if f.injected {
		return false
	}
	f.injected = true